pennylane/operation.py

```diff
@@ -800,6 +800,97 @@ def _matrix(cls, *params):
         return np.diag(cls._eigvals(*params))
 
 
+class Channel(Operation):
+    r"""Base class for quantum channels.
+
+    As with :class:`~.Operation`, the following class attributes must be
+    defined for all channels:
+
+    * :attr:`~.Operator.num_params`
+    * :attr:`~.Operator.num_wires`
+    * :attr:`~.Operator.par_domain`
+
+    The following two class attributes are optional, but in most cases
+    should be clearly defined to avoid unexpected behavior during
+    differentiation.
+
+    * :attr:`~.Operation.grad_method`
+    * :attr:`~.Operation.grad_recipe`
+
+    To define a noisy channel, the following attribute of :class:`~.Channel`
+    can be used to list the corresponding Kraus matrices.
+
+    * :attr:`~.Channel._kraus_matrices`
+
+    Args:
+        params (tuple[float, int, array, Variable]): operation parameters
+
+    Keyword Args:
+        wires (Sequence[int]): Subsystems the channel acts on. If not given, args[-1]
+            is interpreted as wires.
+        do_queue (bool): Indicates whether the operation should be
+            immediately pushed into a :class:`BaseQNode` circuit queue.
+            This flag is useful if there is some reason to run an Operation
+            outside of a BaseQNode context.
+    """
+    # pylint: disable=abstract-method
+
+    @classmethod
+    def _kraus_matrices(cls, *params):
+        """Kraus matrices representing a quantum channel, specified in
+        the computational basis.
+
+        This is a class method that should be defined for all
+        new channels. It returns the Kraus matrices representing
+        the channel in the computational basis.
```

Comment:
If this is so, would it be worth making it an `abstractmethod`? That way it can be enforced that an error is raised for every subclass of `Channel`.

> (..) cannot be instantiated unless all of its abstract methods and properties are overridden

[(part of `abstractmethod`)](https://docs.python.org/3/library/abc.html#abc.abstractmethod)

Reply:
yes, that makes a lot of sense! Thanks! :)